tests/functional/test_uninstall.py

```diff
@@ -166,20 +169,47 @@ def test_uninstall_entry_point(script):
         setup(
             name='ep-install',
             version='0.1',
-            entry_points={"pip_test.ep":
-                          ["ep:name1 = distutils_install",
-                           "ep:name2 = distutils_install"]
-                          }
+            entry_points={{"console_scripts": ["{0}", ],
+                           "pip_test.ep":
+                           ["ep:name1 = distutils_install",
+                            "ep:name2 = distutils_install"]
+                          }}
         )
-    """))
+    """.format(console_scripts)))
+    script_name = script.bin_path.join(console_scripts.split('=')[0].strip())
     result = script.pip('install', pkg_path)
+    assert script_name.exists
     result = script.pip('list', '--format=legacy')
     assert "ep-install (0.1)" in result.stdout
     script.pip('uninstall', 'ep_install', '-y')
+    assert not script_name.exists
     result2 = script.pip('list', '--format=legacy')
     assert "ep-install (0.1)" not in result2.stdout
 
 
+def test_uninstall_gui_scripts(script):
+    """
+    Make sure that uninstall removes gui scripts
+    """
+    pkg_name = "gui_pkg"
+    script.scratch_path.join(pkg_name).mkdir()
+    pkg_path = script.scratch_path / pkg_name
+    pkg_path.join("setup.py").write(textwrap.dedent("""
+        from setuptools import setup
+        setup(
+            name='{0}',
+            version='0.1',
+            entry_points={{"gui_scripts": ["test_ = distutils_install", ],
+                          }}
+        )
+    """.format(pkg_name)))
```

Comment:
I'm working on it.
However not hat I have rebased for some reason I cannot run ``tox`` on my machine. I get errors like

> tox/py35/bin/python: Error while finding spec for 'pip.__main__' (<class 'ImportError'>: bad magic number in 'pip._vendor.html5lib.serializer': b'\x03\xf3\r\n'); 'pip' is 
a package and cannot be directly executed

I'll see what the CIs tell me and act accordingly